packages/internal/src/build/api.ts

```diff
@@ -108,21 +108,26 @@ export const prebuildApiFiles = (srcFiles: string[]) => {
   const plugins = getApiSideBabelPlugins()
 
   return srcFiles.map((srcPath) => {
+    let fileContent
     const relativePathFromSrc = path.relative(rwjsPaths.base, srcPath)
     const dstPath = path
       .join(rwjsPaths.generated.prebuild, relativePathFromSrc)
       .replace(/\.(ts)$/, '.js')
 
-    const result = prebuildApiFile(srcPath, dstPath, plugins)
-    if (!result?.code) {
-      // TODO: Figure out a better way to return these programatically.
-      console.warn('Error:', srcPath, 'could not prebuilt.')
-
-      return undefined
+    if (srcPath.endsWith('.json')) {
+      fileContent = fs.readFileSync(srcPath, 'utf-8')
+    } else {
+      const result = prebuildApiFile(srcPath, dstPath, plugins)
+      if (!result?.code) {
+        // TODO: Figure out a better way to return these programatically.
+        console.warn('Error:', srcPath, 'could not prebuilt.')
+
+        return undefined
+      }
+      fileContent = result.code
     }
```

Comment:
Nice work diving into this @aguscha333! 🚀

If we do add this functionality, we probably want to leave this function alone for now. The way this function's written, it's only concerned with JS/TS files. If we started supporting JSON generically, it'd make sense to tweak this, but since we're only really copying one file over, instead of retrofitting it to handle JSON, I feel like we could just add a bit of logic to the `buildApi` function to copy the `datamodel.json` if it exists:

```diff
export const buildApi = () => {
  // TODO: Be smarter about caching and invalidating files,
  // but right now we just delete everything.
  cleanApiBuild()

  const srcFiles = findApiFiles()

  const prebuiltFiles = prebuildApiFiles(srcFiles)
    .filter((path): path is string => path !== undefined)
    .flatMap(generateProxyFilesForNestedFunction)

-  return transpileApi(prebuiltFiles)
+ // Save the return for later
+  const transpileResult = transpileApi(prebuiltFiles)
+
+ // Just copy over `datamodel.json` if it exists at the right path
+  const DATAMODEL_PATH = path.join(getPaths().api.models, 'datamodel.json')
+
+  if (fs.existsSync(DATAMODEL_PATH)) {
+    fs.copyFileSync(
+      DATAMODEL_PATH,
+      path.join(getPaths().api.dist, 'models', 'datamodel.json')
+    )
+  }
+
+  return transpileResult
}
```

The reason I'm trying to make this change a little more minimal is that we're close to a V1 release, so we have to be really careful if we're tweaking the build process. But what do you think?